### ISSUE TITLE
Proof of Revenue Model required for App Mining Apps

### ISSUE THREAD

**Issue by joshzensite** [blockstack/app-mining#136]
Currently, the rank is determined by :
1. Credible upvotes (PH) 
2. Usability, Usefulness, Credibility and Desirability (TryMyUI)
3. Auth Method & GAIA (NIL) 
4. Impression, Mentions & Growth (Awario) 

Above are the current method used to measure how successful an app is to earn the highest placement in App Mining. 

These are the current issues:

1. New Blockstack apps that are younger than 1.5 months being able to have a placement.
2. Older apps with good ranking having no business model in place. 

Currently, based on the Circular paper, there is 15 reward period. And, there are 5 apps without a revenue model in the top 8 placement. 

What happens when the reward period comes to an end? 

I understand no businesses make money on the first day. That is why I would suggest:

For New Submission: 
1. Business Lean Canvas required for submission.

Reason: Regardless of centralization or decentralization, every product should be self-sustaining in terms of revenue. 

For Apps older than 3 months : 
1. Proof of revenue generation. 
2. The growth rate in the first 3 months. 

Reason: Revenue is the best metric to prove the usefulness of a product and the prize money could be useful to other builders/founders during very early stages. 

Reason 2: Whenever a product gets demoted from higher placement, there is a noticeable drop-in activity and progress. (Motivation is no longer there.) It is better to have 10 successful apps built on Blockstack that ended up becoming big and famous rather than 10,000 failed apps. 

Reason 3: Revenue model is also where STX can be earned and spent by anyone. High circulation is good for the token itself.

**Comment by friedger** [comment]
That could be handled by a reviewer with binary score.

However, if the team provides a business model and get bought by one of the
big tech companies then something went wrong.

I'd say a successful dapp have many clones, many forks, many nodes on their
app chain, and most importantly has a model where service providers are
service consumers, i.e. there is a benefit for the user in contributing to
the app.

Here might be a good source for "better" models:
https://github.com/FEMBusinessModelsRing/web3_revenue_primitives

**Comment by GinaAbrams** [comment]
Thanks @joshzensite. Who would you propose as an app reviewer for this? 

A robust proof of revenue reviewer would measure revenue growth versus just having a revenue plan. It could be early in app mining for this to be implemented.

**Comment by joshzensite** [comment]
I am purposing that every app that wishes participate in the app mining program must present a business lean canvas. It will serve as an intent to be profitable in the medium / long run. This is to ensure Blockstack apps can sustain without App Mining Program. Blockstack may be the initial reviewer for this initiative. 

Sample of BLC : 
![15064249490516_Deezer](https://user-images.githubusercontent.com/45140771/62671815-d59ed780-b9ca-11e9-951a-10c4912a5ba3.png)

There is no better way to put this, awarding apps that are not independent is a waste of resources. 

P.S: Something is really wrong with my English.

**Comment by dantrevino** [comment]
This is anti-free software.  The implication that only profit-seeking applications are valuable is a wrong and ignores that last 25+ years of software innovation.

**Comment by sdsantos** [comment]
@dantrevino I also thought about open source. What about either having a business model or being fully open source?

**Comment by dantrevino** [comment]
@sdsantos I personally, would argue against that as well.  In my opinion, we should not put any constraints on what software is being incentivized.  There is a place, really a *need*,  for a broad spectrum of applications.  The central question that this proposal brings to me is:

Does this community only care about 1 or 2 types/categories of software?  Or do we care about the *quality* of the software?

Its a good question, worthy of discussion, but I've always argued that the 'killer app' for bringing people to user-focused (decentralized) applications is a broad, vibrant, ecosystem of applications.  Not a single app, and certainly not a single category (paid, free, subscription, whatever) of apps.

**Comment by joshzensite** [comment]
@dantrevino I agree there is some flaws with my proposal, however, we also need to have a good ratio between FREE and PAID.

**Comment by friedger** [comment]
Still to be seen how PAID looks like in the decentralised world

**Comment by jeffdomke** [comment]
We discussed this on the PBC team a couple times. Directionally agree that apps generating revenue is a positive thing, but also agree that forcing all apps to do so doesn't make sense. Even if it did we don't have a reliable external reviewer for this. The plan for revenue is trivial, what we really want is a reviewer to measure revenue which is challenging. Until we have more details and a proposed reviewer there isn't much to discuss. Moving this to review and assuming it will be closed.

**Comment by larrysalibra** [comment]
I wrote up some thoughts on how we can measure value delivered to users by apps here based on a call with @pstan26  https://forum.blockstack.org/t/measuring-user-value-hodling-stx-and-one-click-payments/9418?u=larry

**Comment by larrysalibra** [comment]
**TL;DR:**

1. reward value apps deliver to users by revenue or profit - do this by partnering with a payment provider
2. reward app miners that HODL STX
3. make a Blockstack “one-click” payment system integrated with auth, identity & storage (via @pstan26 )

We’ve clearly heard from app miners that they'd like to see metrics that are more closely aligned with the value apps provide to users. We've talked about measuring user growth/retention, but this raises a number of questions such as how to define a user, how to prevent Sybil attacks, is one user the same value as another user etc. It also involves tracking users which is against our values. 

We've also discussed [tracking app progress](https://github.com/blockstack/app-mining/issues/174): are app miners continuing to work on their apps, adding additional features, etc. This also has a bunch of problems: it’s a very subjective measure, features may or may not add value to users, etc.

Today, Patrick and I had a great chat where we honed in on the idea that what we’re really trying to measure here is **the value apps provide to users**. We want more apps that users find useful (valuable) and less of apps from which users derive little or no value. If we’re able to measure this, we can neatly sidestep all of these other questions. We don’t need to worry about how apps provide value as long as they do it in a way that conforms with our Can’t Be Evil principles. 

There’s [no need to reinvent the wheel on this](https://github.com/blockstack/app-mining/issues/136) - there’s already a widely used method of measure the value delivered to users: **how much profit a company makes** and how much of that comes from money paid by users to the company.

There are a few ways we can do that. The traditional way of getting a relatively objective view of the financials of a company is for the company to prepare periodic financial statements and have those audited by independent auditor.
While it isn’t realistic for us to ask apps to prepare financial statements on a monthly or likely even quarterly basis, it’s something we might consider on a biannual or annual basis. Companies could be ranked any metric or metrics we like such as net income, gross income or retained earnings. The downside of this is that it would be time and money intensive for all parties involved. A metric that changes only a couple times per year is also not the best for our fast moving industry.

Another method of achieving almost the same result that involves less human involves fewer moving parts, would be to partner with one or more payment processors - Stripe or Paypal for example - and have app developers connect their payment processor accounts with a platform we put together that monitors aggregate customer payment (no individual customer info of course). We could then score apps based on gross revenue. 

My view is that app developers should be trying to build sustainable businesses around their apps. The way to sustain a user-centric, Can’t Be Evil business is to charge the user for the value they get from their apps.

If app miners can figure out how to make something that users find valuable and are willing to pay for, and do it in a way that Can’t Be Evil, all of the other pieces: UI, UX, marketing in promotion, app maintenance, etc will fall into place.

By partnering with a payment provider and building Blockstack-friendly tools around it, @pstan26  had an idea that I think is worth exploring that we could create an Blockstack-friendly one-click payment mechanism. 

You might ask, why use a traditional fiat payment processor? Why not force apps to accept STX or BTC? If I’ve learned anything from my 5+ years involved in the crypto space, it’s that users aren’t really that interested in paying for things with crypto **when there’s a lower friction, higher reward option.** 
Our goal with Blockstack is not to popularize crypto as a payment method but to build an ecosystem of apps that can’t be evil. In that ecosystem, STX are the necessarily fuel that powers the infrastructure. Forcing crypto on app end users as the *only* payment method when there are payment methods that are widely adopted, lower cost and have a much better user experience is a recipe for making sure your app gets no user adoption.  App devs can use their own stash of STX to pay for names and other digital assets and transfer those assets to the users - users don’t need to make these crypto payments themselves.

In a world where successful app miners are earning fiat revenue from their users, they have less need to immediately liquidate STX tokens to fund their operations. Instead, they can take a longer-term view of building a successful app ecosystem. They can afford to HODL their STX and those that do should get rewarded for doing so and supporting the growth of the ecosystem. We could introduce HODLing of STX tokens as an additional app mining metric. App Miners that HODL their STX would get an additional bump in their app mining score.

I'd love to hear your thoughts!

See @cuevasm's thoughts here: https://forum.blockstack.org/t/measuring-user-value-hodling-stx-and-one-click-payments/9418/2?u=larry
